core/src/main/scala/org/scalamock/function/FakeFunction.scala

```diff
@@ -37,9 +37,14 @@ abstract class FakeFunction(protected val mockContext: MockContext, private[scal
   private def expectationContext = mockContext.expectationContext
 
   def handle(arguments: Product): Any = {
-    val call = new Call(this, arguments)
-    callLog += call
-    expectationContext.handle(call) getOrElse onUnexpected(call)
+    if (Option(callLog).isDefined) {
```

Comment:
I would better check `!= null` or change code to prevent `null` because this fancy null checking is exaggerated.